### ISSUE TITLE
Video audio only playing through the earpiece on iOS once audio is recorded and played

### ISSUE THREAD

**Issue by tslater** [expo/expo#9915]
### Summary of Issue
On iOS, once audio is recorded and played back, the video no longer plays through the bottom speaker, only through the ear piece.

### Environment - output of `expo diagnostics` & the platform(s) you're targeting
iOS 13.6.1 iPhone SE (2020)

Expo diagnostics:
```
  Expo CLI 3.24.2 environment info:
    System:
      OS: macOS 10.15.4
      Shell: 5.7.1 - /bin/zsh
    Binaries:
      Node: 14.8.0 - ~/.nvm/versions/node/v14.8.0/bin/node
      Yarn: 1.22.4 - /usr/local/bin/yarn
      npm: 6.14.8 - ~/.nvm/versions/node/v14.8.0/bin/npm
      Watchman: 4.9.0 - /usr/local/bin/watchman
    IDEs:
      Android Studio: 4.0 AI-193.6911.18.40.6514223
      Xcode: 11.6/11E708 - /usr/bin/xcodebuild
    npmPackages:
      expo: ^38.0.9 => 38.0.9 
      react: 16.11.0 => 16.11.0 
      react-native: https://github.com/expo/react-native/archive/sdk-38.0.0.tar.gz => 0.62.2 
      react-navigation: ^3.13.0 => 3.13.0 
    npmGlobalPackages:
      expo-cli: 3.24.2
```

#### Offending example:
https://snack.expo.io/OZJ!b7qQe
Note that if I delete the recorder component after using it, the issue persists.

#### Working example (missing the recording, but can help you hear what the video should sound like)
 https://snack.expo.io/zkM2VySDl

### Steps to Reproduce
Basically if you record audio and play it back in an iOS app. Then play a video.

### Expected Behavior vs Actual Behavior

I expect to have audio play through both the bottom speaker and the earpiece. It is only in the ear piece.

**Comment by tslater** [comment]
In my searching, I believe this person had the same issue:
https://forums.expo.io/t/trying-to-use-expo-audio-to-record-and-play-sound-at-the-same-time/30216.

It is a very stateful error, even after removing the audio recording code and restarting restarting the app, the audio is permanently relegated to the earpiece.

**Comment by IjzerenHein** [comment]
Hi! I'm unable to reproduce the problem (tried on an iPhone XR using Airpods). When I record and stop the recording, the video sound keeps playing from the bottom speaker.

Can you record a screen video demonstrating the steps that you take?
Also, which Expo client are you using (I tried with 2.16.0)

**Comment by kevgrig** [comment]
@IjzerenHein 
> I'm unable to reproduce the problem (tried on an iPhone XR using Airpods).

I just reported a similar issue in #11246 and I note above that you say you tried using Airpods. Can you please try using the phone speakers instead? In my issue, it only occurs when trying to use the phone speaker (headphones work fine).

**Comment by MrXyfir** [comment]
Any solution to this?

**Comment by aneequeahmad** [comment]
still the issue exists.

**Comment by MrXyfir** [comment]
Luckily if you do `Audio.setAudioModeAsync({ allowsRecordingIOS: false })` after recording it'll work as expected again but it's still frustrating.

**Comment by IjzerenHein** [comment]
I think @MrXyfir has a very good point. This might just be related to the Audio-mode that is set when calling `Audio.setAudioModeAsync({ allowsRecordingIOS: false })`. This sets the audio up for both recording and audio playback at the same time (AVSession -> AVAudioSessionCategoryPlayAndRecord). Details here:
https://developer.apple.com/documentation/avfaudio/avaudiosessioncategoryplayandrecord?language=objc

In this mode audio is played at a lower volume and might be routed to the headphones.

Try setting `allowsRecordingIOS` to `true` before recording, and to `false` after recording has completed.

**Comment by MrXyfir** [comment]
Still an issue.

**Comment by dorthwein** [comment]
We're running into this as well using a different package to stream audio recordings

**Comment by dorthwein** [comment]
Is there any way to put a bounty on this?

**Comment by tslater** [comment]
@dorthwein I actually think this _might_ not be an expo issue. I think it is a way with how iOS handles audio modes with recording. We solved the problem by just explicitly setting the audio mode in our video/audio players in on the component mount:

```
import { Audio } from 'expo-av';

....


  useEffect(() => {
    // make sure the sound plays...and not in the earpiece like a phone
    Audio.setAudioModeAsync(audioModePlayback)

  }, [])
```

Of course if your playing and recording interfaces are concurrently mounted, you would probably want to set the audio mode either before every playback, or maybe after every recording event is over?

**Comment by LuisNeves05** [comment]
> I think @MrXyfir has a very good point. This might just be related to the Audio-mode that is set when calling `Audio.setAudioModeAsync({ allowsRecordingIOS: false })`. This sets the audio up for both recording and audio playback at the same time (AVSession -> AVAudioSessionCategoryPlayAndRecord). Details here: https://developer.apple.com/documentation/avfaudio/avaudiosessioncategoryplayandrecord?language=objc
> 
> In this mode audio is played at a lower volume and might be routed to the headphones.
> 
> Try setting `allowsRecordingIOS` to `true` before recording, and to `false` after recording has completed.

Your sir, just saved me a LOT of time. Thank you

**Comment by stevehs17** [comment]
Still an issue.

**Comment by zero19124** [comment]
this is working 

 Audio.setAudioModeAsync({ allowsRecordingIOS: false })
    const { sound } = await Audio.Sound.createAsync({
      uri: text,
    });

**Comment by aramis-it** [comment]
Solution same in expo-audio and in 2025?

**Comment by alnotara** [comment]
> Solution same in expo-audio and in 2025?

I'm also scouring the internet regarding this. I'm unable to find a way to use `expo-audio` and have the recorded audio playback via the iPhone's speaker as opposed to the earpiece. 

Any help would be much appreciated!

**Comment by MrXyfir** [comment]
All the info needed should be up above. Essentially you just need to switch the audio mode away from recording and back to playback. I don't know why Expo doesn't just handle the switch automatically, but it seems to be a deliberate choice and not a bug.

**Comment by alnotara** [comment]
@MrXyfir thanks for getting back to me. I believe those solutions are all using `expo-av`, which is deprecated and replaced with `expo-audio`. I'm not seeing the same configurations exposed in the latter package, and I'm assuming they didn't automate it either as I'm seeing the same issues with the recording playback coming through the earpiece. 

Wondering if there is a solution for `expo-audio` out there somewhere?

**Comment by MrXyfir** [comment]
@alnotara I haven't worked with this in a while so I'm not certain but it looks like [`Audio.setAudioModeAsync(mode)`](https://docs.expo.dev/versions/latest/sdk/audio/#audiosetaudiomodeasyncmode) is probably what you need. Just make sure to disable recording mode after recording and then playback should work correctly again.

**Comment by lucassimao** [comment]
I finally managed to fix this by calling await setAudioModeAsync({ allowsRecording: false }); right before playing anything with expo-audio

```javascript
  const playAudio = async () => {

    await setAudioModeAsync({ allowsRecording: false });

    try {
      if (isPlaying) {
        player.pause();
        player.seekTo(0);
      } else {
        player.play();
      }
    } catch (error) {
      console.error("Error playing audio:", error);
    }
  };
```

In my case, the problem happened right after finishing a webrtc session (using [react-native-webrtc](https://github.com/react-native-webrtc/react-native-webrtc)
